crates/ra_hir_ty/src/expr.rs

```diff
@@ -501,4 +490,32 @@ mod tests {
             ",
         );
     }
+
+    #[test]
+    fn tuple_struct() {
+        check_diagnostic(
+            r"
+            struct Tup(u8, u16);
+            fn f() {
+                Tup(0);
+            }
+            ",
+            expect![["\"Tup(0)\": Expected 2 arguments, found 1\n"]],
```

Comment:
Actually now that I look at it, annotaed tests might be even better here

```
            struct Tup(u8, u16);
            fn f() {
                Tup(0);
              //^^^^^^ Expected 2 arguments, found 1
            }
```

but maybe not -- multiline spans would be awkward. See `inlay_hints.rs` for how such tests feel